### PR TITLE
 Closes #10 -  Add Sign In page REST API.; Closes #24 - Add authification guard

### LINKED ISSUE #10: Add Sign In page REST API.

### LINKED ISSUE #24: Add authification guard

### PR DESCRIPTION
**Pull request by oleksandrlytvyn1223**
In this commit i have added authification guard, added connection to backend added a few directories to navigation bar and created a folders for them so we can see a content for signed in user 
default

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -23,18 +32,37 @@ import { CompareValidatorDirective } from './compare-validator.directive';
...
     FormsModule,
-    ReactiveFormsModule
+    ReactiveFormsModule,
+    HttpClientModule,
+    JwtModule.forRoot({
+      config: {
+        tokenGetter: function  tokenGetter() {
+             return     localStorage.getItem('access_token');},
+        whitelistedDomains: [
+          'http://localhost:4200',
```
Comment (roman-fay) [inline src/app/app.module.ts]: create ticket to change to proper address after hosting is done

File: src/app/jwt.service.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+@Injectable({
+providedIn: 'root'
+})
+
+export class JwtService {
+    constructor(private httpClient: HttpClient) { }
+
+  signin(email:string, password:string) {
+      return this.httpClient.post<{access_token:  string}>('https://localhost:44398/api/LogIn',{email, password} )
```
Comment (roman-fay) [inline src/app/jwt.service.ts]: this should not be inline. Some sort of 'Config service' would be a good solution.

File: src/app/signin/signin.component.ts [hunk 3]
```diff
@@ -37,4 +49,28 @@ export class SigninComponent implements OnInit {
...
+
+  onSubmit() {
+    this.submitted = true;
+
+    // stop here if form is invalid
+    if (this.signinform.invalid) {
+        return;
+    }
+    this.loading = true;
+    localStorage.setItem('email',this.f.email.value);
```
Comment (roman-fay) [inline src/app/signin/signin.component.ts]: do we need to store this? and even if we do - wouldn't it be better to save it after sign in?

Comment (roman-fay) [reply src/app/signin/signin.component.ts]: in sign in service for example

File: src/app/app.module.ts [hunk 4]
```diff
@@ -23,18 +32,37 @@ import { CompareValidatorDirective } from './compare-validator.directive';
...
     AppRoutingModule,
     FormsModule,
-    ReactiveFormsModule
+    ReactiveFormsModule,
+    HttpClientModule,
+    JwtModule.forRoot({
+      config: {
+        tokenGetter: function  tokenGetter() {
+             return     localStorage.getItem('access_token');},
+        whitelistedDomains: [
```
Comment (koldovsky) [inline src/app/app.module.ts]: move to environments

File: tsconfig.json [hunk 5]
```diff
@@ -9,6 +9,7 @@
     "moduleResolution": "node",
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
+    "allowJs": true,
```
Comment (koldovsky) [inline tsconfig.json]: Please explain why you need this option

File: src/app/app.module.ts (hunk 4 above)
Comment (oleksandrlytvyn1223) [reply src/app/app.module.ts]: @koldovsky can you explain what you mean? Also currently we dont have much time so i will create ticket for that and push it as it is.

**Pull request merged**